distrib_rl/Experiments/ConfigAdjusters/ParallelAdjuster.py

```diff
@@ -1,13 +1,18 @@
 from distrib_rl.Experiments.ConfigAdjusters import Adjuster
 from distrib_rl.Experiments.ConfigAdjusters import Adjustment
+from distrib_rl.Experiments.ConfigAdjusters.ListAdjuster import ListAdjuster
 
 class ParallelAdjuster(Adjuster):
     def __init__(self):
         super().__init__()
 
     def init(self, adjustments_json, cfg):
         for key, item in adjustments_json.items():
-            if "adjustment" in key:
+            if "list_" in key:
+                adjustment = ListAdjuster()
+                adjustment.init(item, cfg)
+                self.adjustments.append(adjustment)
+            elif "adjustment_" in key:
                 adjustment = Adjustment()
                 adjustment.init(item, cfg)
                 self.adjustments.append(adjustment)
```

Comment:
I tried refactoring this to call out to the `AdujsterFactory`, but of course that introduces a circular dependency. I then went to just do it manually in each of the types, but of course that creates circular dependencies as well.

This is a good example of a case where inversion of control is very helpful, as in IoC, the responsibility for instantiating an object such as this generally resides in a context manager component. Usually on startup a context manager will build up a representation of the desired object dependency tree (with types determined from one of many config sources, including introspection), and then instantiates it from the bottom up.